src/main/scala/diplomacy/LazyModule.scala

```diff
@@ -5,14 +5,104 @@ package freechips.rocketchip.diplomacy
 import Chisel.{defaultCompileOptions => _, _}
 import chisel3.internal.sourceinfo.{SourceInfo, UnlocatableSourceInfo}
 import chisel3.{MultiIOModule, RawModule, Reset, withClockAndReset}
-import chisel3.experimental.{ChiselAnnotation}
+import chisel3.experimental.{BaseModule, ChiselAnnotation}
+import chisel3.aop.{Select => CSelect}
 import firrtl.passes.InlineAnnotation
 import freechips.rocketchip.config.Parameters
 import freechips.rocketchip.util.CompileOptions.NotStrictInferReset
 
 import scala.collection.immutable.{ListMap, SortedMap}
 import scala.util.matching._
 
+object Select {
```

Comment:
we laboriously added scaladoc to this file... please continue this for this object!

Reply:
I would put this in its own file, while also doing as @mwachs5 suggests 😎